### PR TITLE
DeformableConvolution v8 specification

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
 - *Specification of a new DeformableConvolution using a modulation mechanism (https://arxiv.org/abs/1811.11168)*

### Tickets:
 - *52646*

**Formulas:** (rendered from the spec)
DeformableConv v1:
image

DeformableConv v8:
image

**Bilinear interpolation:**

- **DeformableConvolution**:  if the sampling location is outside of the feature map boundary, its sampled value would be zero, the gradient with respect to learnable offset would be zero https://github.com/openvinotoolkit/openvino/blob/e864972f837682fd9a73aa645fc290c4e25d7b7b/inference-engine/src/mkldnn_plugin/nodes/mkldnn_def_conv_node.cpp#L972

- **ModulatedDeformableConvolution**: if the sampling location is within one pixel outside of the feature map boundary, bilinear sampling would also be applied. And gradient with respect to learnable offset can be non zero for such locations.** 
https://github.com/PaddlePaddle/Paddle/blob/bd2a4e23fad95fbd10f5ea33f1ab922b65be6515/paddle/fluid/operators/deformable_conv_func.h#L119

### CODE REVIEW

File: docs/ops/convolution/DeformableConvolution_8.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  * **Description**: *deformable_group* is the number of groups in which *offsets* input and *output* should be split into along the channel axis. Apply the deformable convolution using the i-th part of the offsets part on the i-th out.
+  * **Range of values**: integer value starting from `1`
+  * **Type**: `int`
+  * **Default value**: `1`
+  * **Required**: *no*
+
+* *bilinear_interpolation_padding*
+
+  * **Description**: *bilinear_interpolation_padding* is the number of pixels outside of the feature map boundary to apply bilinear interpolation.
+  * **Range of values**: [0, 1]
```
Comment (itikhono) [inline docs/ops/convolution/DeformableConvolution_8.md:100]: not sure, mb integer value starting from `0`?

File: docs/ops/convolution/DeformableConvolution_8.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
+## DeformableConvolution<a name="DeformableConvolution"></a> {#openvino_docs_ops_convolution_DeformableConvolution_8}
```
Comment (lazarevevgeny) [inline docs/ops/convolution/DeformableConvolution_8.md:1]: Update `ie_docs.xml` file as in the https://github.com/openvinotoolkit/openvino/pull/5773

File: docs/ops/convolution/DeformableConvolution_8.md (hunk 1 above)
Comment (lazarevevgeny) [reply docs/ops/convolution/DeformableConvolution_8.md:102]: I would recommend to not restrict the possible values. Just say that there are not negative.

File: docs/ops/convolution/DeformableConvolution_8.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        <port id="4" precision="FP32">
+            <dim>1</dim>
+            <dim>64</dim>
+            <dim>220</dim>
+            <dim>220</dim>
+        </port>
+    </output>
+</layer>
+```
+
```
Comment (lazarevevgeny) [inline docs/ops/convolution/DeformableConvolution_8.md:169]: nit: remove redundant empty line.

File: docs/ops/convolution/DeformableConvolution_8.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  * **Description**: *bilinear_interpolation_padding* is the number of pixels outside of the feature map boundary to apply bilinear interpolation.
+  * **Range of values**: [0, 1]
+  * **Type**: `int`
+  * **Default value**: `0`
+  * **Required**: *no*
+  
+**Inputs**:
+
+*   **1**: Input tensor of type *T* and rank 4. Layout is `NCYX` (number of batches, number of channels, spatial axes Y and X). Required.
```
Comment (lazarevevgeny) [inline docs/ops/convolution/DeformableConvolution_8.md:109]: Wrap "Required" and "Optional" with double "*" characters.

File: docs/ops/convolution/DeformableConvolution_8.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+* *T*: Any numeric type.
+* *T2*: Any supported floating point.
+ 
+**Example**
+
+2D DeformableConvolution (deformable_group=1)
+```xml
+<layer type="DeformableConvolution" ...>
+    <data dilations="1,1" pads_begin="0,0" pads_end="0,0" strides="1,1" auto_pad="explicit"  group="1" deformable_group="1"/>
```
Comment (lazarevevgeny) [inline docs/ops/convolution/DeformableConvolution_8.md:132]: Redundant space character

File: docs/ops/convolution/DeformableConvolution_8.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+
+**Outputs**:
+
+*   **1**: Output tensor of type *T* and rank 4. Layout is `NOYX` (number of batches, number of kernel output channels, spatial axes Y and X).
+
+**Types**:
+
+* *T*: Any numeric type.
+* *T2*: Any supported floating point.
```
Comment (lazarevevgeny) [inline docs/ops/convolution/DeformableConvolution_8.md:125]: Why the type of the ModulationScalars input is different from other types?

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: docs/ops/convolution/DeformableConvolution_8.md (hunk 6 above)
Comment (itikhono) [reply docs/ops/convolution/DeformableConvolution_8.md:125]: by definition it is a floating number between 0 and 1 inclusive

File: docs/ops/convolution/DeformableConvolution_8.md (hunk 1 above)
Comment (itikhono) [reply docs/ops/convolution/DeformableConvolution_8.md]: done

File: docs/ops/convolution/DeformableConvolution_8.md (hunk 2 above)
Comment (itikhono) [reply docs/ops/convolution/DeformableConvolution_8.md:1]: done

File: docs/ops/convolution/DeformableConvolution_8.md (hunk 3 above)
Comment (itikhono) [reply docs/ops/convolution/DeformableConvolution_8.md]: fixed

File: docs/ops/convolution/DeformableConvolution_8.md (hunk 4 above)
Comment (itikhono) [reply docs/ops/convolution/DeformableConvolution_8.md]: done

File: docs/ops/convolution/DeformableConvolution_8.md (hunk 5 above)
Comment (itikhono) [reply docs/ops/convolution/DeformableConvolution_8.md]: fixed

**Comment by lazarevevgeny** [comment]
@itikhono , please, resolve merge conflicts, so I can merge the PR

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**